grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -802,14 +815,17 @@ class StockMovementService {
             }
         }
 
-        switch(stepNumber) {
-            case "2":
+        switch(OutboundWorkflowState.fromStepNumber(stepNumber)) {
+            case OutboundWorkflowState.ADD_ITEMS:
                 return getAddPageItems(requisition, stockMovementItems)
-            case "4":
+            case OutboundWorkflowState.PICK_ITEMS:
+                if (refresh) {
+                    allocatePicklists(requisition.requisitionItems?.asList())
```

Comment:
Yes! Thank you. This is a really good first step. Am I to assume that getPickPageItems() is now just reading? I know I will get there in a minute, but just wanted to ask this explicitly.

Reply:
yes, after I separated allocation from building the PickPageItem now getPickPage only reads the value

Reply:
The one potential "bummer" of fixing the "side effect" of getPickPageItems() is that Manon is expecting that bug to continue to exist i.e. when you refresh or return to the page it allocates items for any lines that don't have picklist items. So there might need to go back and forther a little with her on that to make sure she feels comfortable with what we did.

Reply:
not a problem, this behavior still remains I am sending getStockMovementItems with `refresh=false` flag after importing the file but after refresh and back forward navigation the flag is on true